scheds/rust/scx_layered/src/bpf/main.bpf.c

```diff
@@ -1208,6 +1213,77 @@ static bool keep_running(struct cpu_ctx *cctx, struct task_struct *p)
 	return false;
 }
 
+bool consume_delayed_dsq(u64 dsq_id, u64 now_sec) {
+	struct task_struct *p;
+	struct task_ctx *tctx;
+	u64 runnable_at_sec;
+	
+	if (!dsq_id || !now_sec)
+		return false;
+	
+	bpf_for_each(scx_dsq, p, dsq_id, 0) {
+		if ((tctx = lookup_task_ctx(p))) {
+			if (tctx->runnable_at) {
+				runnable_at_sec = tctx->runnable_at / CONFIG_HZ;
+				if (now_sec - runnable_at_sec > antistall_sec) {
+					if (scx_bpf_consume(dsq_id))
+						return true;
+				} else {
+					// if the first entry in a dsq is ok, the dsq is ok.
+					return false;
+				}
+			}
+		}
+	}
+	return false;
+}
+
+struct {
+	__uint(type, BPF_MAP_TYPE_ARRAY);
+	__type(key, s32);
+	__type(value, bool);
+	__uint(max_entries, 4096);
+} antistall_cpus SEC(".maps");
```

Comment:
I'd use `BPF_MAP_TYPE_PERCPU_ARRAY` which contains one bool element.

Reply:
I wanted to do this but I did not see a way to write to `BPF_MAP_TYPE_PERCPU_ARRAY` from another CPU (i.e. I saw a way to read from other CPUs, but not write).

I also didn't see anything on timers indicating that they are per cpu, so I figured the timer is running on one CPU, and that CPU needs to write to the maps for other CPUs.

Is my understanding wrong?